recipe/meta.yaml

```diff
@@ -138,18 +138,25 @@ outputs:
       requires:
         - pip
       commands:
-        - mamba --help
-        - if exist %PREFIX%\condabin\mamba.bat (exit 0) else (exit 1)  # [win]
-        - test -f ${PREFIX}/etc/profile.d/mamba.sh  # [linux]
-        # these tests work when run on win, but for some reason not during conda build
-        - mamba create -n test_py2 python=2.7 --dry-run  # [unix]
-        - mamba clean --all --dry-run  # [unix]
-        - mamba repoquery whoneeds conda  # [unix]
-        - mamba install xtensor xsimd -c conda-forge --dry-run  # [linux and x86_64]
+        - test -f ${PREFIX}/condabin/mamba  # [unix]
+        - test -f ${PREFIX}/etc/profile.d/mamba.sh  # [unix]
+        - if not exist %PREFIX%\condabin\mamba.bat exit 1  # [win]
+
+        # Ensure we run the executable from the prefix and not from the base installation.
+        - test "$(command -v mamba)" = "${PREFIX}/bin/mamba"  # [unix]
+        - cmd /q /c "for /f "delims=" %%f in ('C:\Windows\System32\where mamba.bat') do if "%%~dpf"=="%LIBRARY_BIN%\" (exit 0) else (exit 1)"  # [win]
```

Comment:
This cryptic Batch nonsense should be equivalent to the POSIX shell test above it and checks that the first entry on `PATH` with the executable is the one we want to test.
It's wrapped in a `cmd /c` call since we have an `exit 0` in there (to only check the first entry) which would exit the whole test script if it wasn't wrapped in a subprocess.
I tried to use `findstr "/C:%PREFIX%"` in there, but it turns out that the search strings for `findstr` are length-limited far below the `PREFIX`' lengths we use.